pandas/core/indexes/datetimelike.py

```diff
@@ -237,22 +237,21 @@ def min(self, axis=None, skipna=True, *args, **kwargs):
             return self._na_value
 
         i8 = self.asi8
-        try:
-            # quick check
-            if len(i8) and self.is_monotonic:
-                if i8[0] != iNaT:
-                    return self._data._box_func(i8[0])
 
-            if self.hasnans:
-                if skipna:
-                    min_stamp = self[~self._isnan].asi8.min()
-                else:
-                    return self._na_value
-            else:
+        # quick check
+        if self.is_monotonic and i8[0] != iNaT:
+            return self._data._box_func(i8[0])
+        try:
+            if self.hasnans and skipna:
```

Comment:
I suspect you can remove this try/except entirely here

Reply:
It's there because `self[~self._isnan]` might be empty:
```python
(Pdb) self[~self._isnan].asi8
array([], dtype=int64)
(Pdb) self[~self._isnan].asi8.min()
*** ValueError: zero-size array to reduction operation minimum which has no identity
```
I've removed it and put an extra condition